homeassistant/components/sensor/emoncms.py

```diff
@@ -0,0 +1,246 @@
+"""
+Support for monitoring emoncms feeds.
+
+https://emoncms.org
+"""
+from datetime import timedelta
+import logging
+import requests
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util as util
+import homeassistant.util.dt as dt_util
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_API_KEY, CONF_URL, CONF_VALUE_TEMPLATE,
+    CONF_UNIT_OF_MEASUREMENT, CONF_ID, CONF_SCAN_INTERVAL,
+    STATE_UNKNOWN)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers import template
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DECIMALS = 2
+CONF_EXCLUDE_FEEDID = "exclude_feed_id"
+CONF_INCLUDE_FEEDID = "include_feed_id"
+CONF_INCLUDE_FEEDID_NAMES = "include_feed_id_names"
+DEFAULT_SCAN_INTERVAL = 60
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_ID): cv.positive_int,
+    vol.Exclusive(CONF_INCLUDE_FEEDID, 'only_include_or_exclude'):
+        vol.All(cv.ensure_list, [cv.positive_int]),
+    vol.Exclusive(CONF_EXCLUDE_FEEDID, 'only_include_or_exclude'):
+        vol.All(cv.ensure_list, [cv.positive_int]),
+    vol.Optional(CONF_INCLUDE_FEEDID_NAMES):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT, default="W"): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL):
+        vol.All(vol.Coerce(int), vol.Range(min=5)),
+
+})
+
+ATTR_SIZE = 'Size'
+ATTR_LASTUPDATETIME = 'LastUpdated'
+ATTR_TAG = 'Tag'
+ATTR_FEEDID = 'FeedId'
+ATTR_USERID = 'UserId'
+ATTR_FEEDNAME = 'FeedName'
+ATTR_LASTUPDATETIMESTR = 'LastUpdatedStr'
+
+
+def get_id(sensorid, feedtag, feedname, feedid, feeduserid):
+    """Return unique identifier for feed / sensor."""
+    return "emoncms" + sensorid + "_" + feedtag + "_" + \
```

Comment:
Sorry, I've used back-tics around the string above. Those should be regular single quotes or double quotes depending on your preference. Just be consistent over the module.